src/vm/assemblynative.cpp

```diff
@@ -1428,3 +1428,27 @@ FCIMPL0(FC_BOOL_RET, AssemblyNative::IsTracingEnabled)
     FC_RETURN_BOOL(BinderTracing::IsEnabled());
 }
 FCIMPLEND
+
+// static
+void QCALLTYPE AssemblyNative::TraceResolvingHandlerInvoked(LPCWSTR assemblyName, LPCWSTR handlerName, LPCWSTR alcName, LPCWSTR resultAssemblyName, LPCWSTR resultAssemblyPath)
+{
+    QCALL_CONTRACT;
+
+    BEGIN_QCALL;
+
+    FireEtwALCResolvingHandlerInvoked(GetClrInstanceId(), assemblyName, handlerName, alcName, resultAssemblyName, resultAssemblyPath);
```

Comment:
There should be a way to fire these events from managed code via the NativeRuntimeEventSource and avoid the overhead of a QCall.

Reply:
Agreed with the sentiment. I believe though and @elinor-fung can confirm that there are cases where we want this event to fire where we cannot call managed code and the intent here is to have a single way instead of two actual "fire event" paths. If that isn't the case, I would agree avoiding the QCall would be preferred if possible.

Reply:
To be clear here I am merely advocating for the desire to have as much in managed **or** native as possible since it makes debugging simpler. I don't think the performance issue in the logging scenario is as interesting since it is a pay-for-play scenario anyways.

Reply:
> we want this event to fire where we cannot call managed code 

I do not think this should be ever the case.  All places that fire this event should be able to call managed code too.

Reply:
@jkotas Okay. Then @elinor-fung will need to comment. I was just guessing as to a reason for why this might have been done.

Reply:
Unfortunately, the `NativeRuntimeEventSource` is not actually a 'normal' managed EventSource and does not properly support emitting events. After chatting with @sywhang, we went with a QCall. While the overhead is non-ideal, to turn on tracing is to accept some overhead.

> All places that fire this event should be able to call managed code too.

Yes.

Reply:
I want to add that in theory it would be possible to implement managed event firing even for `NativeRuntimeEventSource` but it's a non-trivial effort, with somewhat questionable value - it's actually unclear if it would internally have to do some kind of FCall/QCall anyway to synchronize the two parts anyway - so we would be just making it "nicer", not necessarily faster.

As noted above, we did discuss this with the EventSource owners, and they suggested to go with a simple QCall for now.